File: recursion/code_samples/python/factorial.py
def factorial_number(n):
    '''An example of a recursive factorial function written in python.'''
    try:
        assert n >= 0 and int(n) == n
        if n in [0,1]:
            return 1
        return n * factorial_number(n-1)
    except AssertionError:
        print (f"Factorial Sum must be a positive interger, not {n}.")
        return exit(1)

if __name__ == "__main__":
    x = factorial_number(10)
    print(x)

File: recursion/code_samples/java/factorial.java
package java;

public class factorial {
    static int factorialNumber(int n) {
        if (n < 2){
            return 1;
          } 
        return (n * factorialNumber(n-1));
    }
    public static void main(String[] args) {
        int x = factorialNumber(10);
        System.out.println(x);
    }
}
